Comment Moderation - time based bans

infinite-persistence opened the pull request:
## 🔵 Current Status 🔵
- [x] Regression testing
- [ ] Tom's blessing 

## Dev Notes
https://www.notion.so/lbry/Comment-Timeout-83a2834a888f41be998c91abcd92e908

## Ticket
Closes [#6712 Comment Moderation - time based bans](https://github.com/lbryio/lbry-desktop/issues/6712)

## Approach
- Consolidated the 3 types of blocking buttons in the Comment Content-menu (i.e. _Block_, _Moderator Block_, _Admin Block_) into 1 regular _Block_ button.
- Show a modal when _Block_ is clicked.
    - Let user choose the blocklist (Personal, Moderator, Admin). 
        - Most people will just have Personal
    - Let user choose the timeout duration (this PR's impetus).

<img src="https://user-images.githubusercontent.com/64950861/130548077-7811d65d-f609-4475-a586-266840066218.png" width="400">

tzarebczan commented:
Checking with Mark based on your notes. I also had a time based ban not work, he's looking into it. 

Aside from blocked list showing the time based ban, can also show it on the channel page where we show a channel is blocked.

infinite-persistence commented:
#### 3ea1b41 to 520a155
Minor tweaks and improvements:
- Fix bug for other block buttons outside of comments (note: they all do "permanent block" for now, no timeout capability).
- hide tab if "Personal" is the only choice.
- add persistence to the tab, type and duration settings.
- minor layout tweaks and mobile fixes.
- block floats

infinite-persistence commented:
### 520a155 to 30c19fe
- Match Commentron's update of the resolution to 1s.
- Changed to Twitch-style of defining the ban duration, i.e. "2d, 1h, 15m".  It is easier to code (as long as the plugin is secure, fingers crossed), and the model is well vetted via Twitch.  Added a help "?" to give examples, regardless.
- Display the timeout info (date, duration, remaining info) in the Blocklist Page.

kauffj commented on ui/modal/modalBlockChannel/view.jsx:
```diff
@@ -0,0 +1,286 @@
+
+    if (Number.isInteger(seconds) && seconds > 0) {
+      if (seconds > ONE_YEAR_IN_SECONDS) {
+        setInvalid('Wow, banned for more than 100 years?');
```
inconsistent

infinite-persistence replied:
As in, not formal?

infinite-persistence commented:
### 30c19fe to c6237a6
- Fix typo in constant naming
- Rebased to resolve conflict

infinite-persistence commented:
## a49db3c to bf8ab2e 
- Fix bug of potentially calling the moderator API instead of the personal API due to persisted state.  It now calls the correct api for the case when the modal is not needed.
